feature/experimental/bgp/ate_tests/bgp_long_lived_graceful_restart/bgp_long_lived_graceful_restart_test.go

```diff
@@ -0,0 +1,1339 @@
+		gnmi.Delete(t, dut, dutConfPath.Neighbor(nbr.neighborip).Config())
+	}
+	fptest.LogQuery(t, "DUT BGP Config", dutConfPath.Config(), gnmi.GetConfig(t, dut, dutConfPath.Config()))
+}
+
+func restartRoutingProcess(t *testing.T, dut *ondatra.DUTDevice) {
+	if _, ok := routingDaemon[dut.Vendor()]; !ok {
+		t.Fatalf("Please add support for vendor %v in var routingDaemon", dut.Vendor())
+	}
+	t.Run("KillGRIBIDaemon", func(t *testing.T) {
+		// Find the PID of routing Daemon.
+		var pId uint64
+		pName := routingDaemon[dut.Vendor()]
+		t.Run("FindroutingDaemonPid", func(t *testing.T) {
+			pId = findProcessByName(t, dut, pName)
+			if pId == 0 {
+				t.Fatalf("Couldn't find pid of routing daemon '%s'", pName)
+			} else {
+				t.Logf("Pid of routing daemon '%s' is '%d'", pName, pId)
+			}
+		})
+
+		// Kill routing daemon through gNOI Kill Request.
+		t.Run("ExecuteGnoiKill", func(t *testing.T) {
+			// TODO - pid type is uint64 in oc-system model, but uint32 in gNOI Kill Request proto.
+			// Until the models are brought in line, typecasting the uint64 to uint32.
+			gNOIKillProcess(t, dut, pName, uint32(pId))
+			// Wait for a bit for routing daemon on the DUT to restart.
+			time.Sleep(30 * time.Second)
+		})
+	})
+}
+
+// findProcessByName uses telemetry to find out the PID of a process
+func findProcessByName(t *testing.T, dut *ondatra.DUTDevice, pName string) uint64 {
+	pList := gnmi.GetAll(t, dut, gnmi.OC().System().ProcessAny().State())
+	var pID uint64
+	for _, proc := range pList {
+		if proc.GetName() == pName {
+			pID = proc.GetPid()
+			t.Logf("Pid of daemon '%s' is '%d'", pName, pID)
+		}
+	}
+	return pID
+}
+
+// gNOIKillProcess kills a daemon on the DUT, given its name and pid.
+func gNOIKillProcess(t *testing.T, dut *ondatra.DUTDevice, pName string, pID uint32) {
+	gnoiClient := dut.RawAPIs().GNOI().Default(t)
+	killRequest := &gnps.KillProcessRequest{Name: pName, Pid: pID, Signal: gnps.KillProcessRequest_SIGNAL_TERM, Restart: true}
+	killResponse, err := gnoiClient.System().KillProcess(context.Background(), killRequest)
+	t.Logf("Got kill process response: %v\n\n", killResponse)
+	if err != nil {
+		t.Fatalf("Failed to execute gNOI Kill Process, error received: %v", err)
+	}
+}
+
+// setBgpPolicy is used to configure routing policy on DUT.
+func setBgpPolicy(t *testing.T, dut *ondatra.DUTDevice, d *oc.Root) {
+	// Configure SetMED on DUT.
+	rp := d.GetOrCreateRoutingPolicy()
+	pdef5 := rp.GetOrCreatePolicyDefinition(setALLOWPolicy)
+	actions5 := pdef5.GetOrCreateStatement(aclStatement3).GetOrCreateActions()
+	actions5.GetOrCreateBgpActions().SetMed = oc.UnionUint32(bgpMED)
+	actions5.GetOrCreateBgpActions().SetLocalPref = ygot.Uint32(100)
+	gnmi.Update(t, dut, gnmi.OC().RoutingPolicy().Config(), rp)
+	//gnmi.Replace(t, dut, gnmi.OC().RoutingPolicy().Config(), rp)
+}
+
+// configureDutNewPeers configured five more BGP peers on subinterfaces.
+func configureDutNewPeers(t *testing.T, dut *ondatra.DUTDevice, nbrs []*bgpNeighbor) {
```

Comment:
"Dut" needs to be "DUT" here -- go initialism style rule

Reply:
updated.